### PR TITLE
vagrant: Add Vagrant environment

### PR DESCRIPTION
**Pull request by mestery**
Add a Vagrant environment, which makes testing a bit easier. Instructions are added on how to use the environment.

Note, I had to modify a few scripts to allow for integration with the Vagrant environment.

Closes Issue #36

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

**Comment by stolsma** [comment]
Nice! Will try this next week on my bare metal Supermicro Mini Server! Does thus need adjustments because of merged #32 ?

**Comment by mestery** [comment]
@stolsma I’m on vacation this week, I’ll look for feedback from you next week and verify any conflicts from #32 as well.

**Comment by mestery** [comment]
@stolsma I just rebased my change on top of #32, should be good to go now.

**Comment by stolsma** [comment]
Will do a last install tomorrow morning (CET)!

**Comment by mestery** [comment]
@n-sandeep If you're ok with this pull request enabling a Vagrant VM where P4-OVS runs natively in the VM and not in a docker container, can we merge this one please? It makes using this easier in terms of spinning the test environment up.

**Comment by n-sandeep** [comment]
> @n-sandeep If you're ok with this pull request enabling a Vagrant VM where P4-OVS runs natively in the VM and not in a docker container, can we merge this one please? It makes using this easier in terms of spinning the test environment up.

Hello Kyle,

Thank you, I would also request @vsureshkumarp and @Namrata-intel to comment on this.
I am a novice in Vagrant..!!!

**Comment by mestery** [comment]
I will push another commit to this PR later today which adds a nice script to bringup a couple of nested VMs, which makes testing out what is here much easier. Still debugging why P4-OVS doesn't work in the environment, but getting closer.

File: build/IPDK_Container/scripts/rundemo.sh
Comment (stolsma) [inline build/IPDK_Container/scripts/rundemo.sh:44]: I don't understand... the get-image script you added downloads a ubuntu qcow2 file and converts it to raw and directly after using the script the file is converted back to qcow2 again?

File: build/IPDK_Container/scripts/rundemo.sh
Comment (stolsma) [inline build/IPDK_Container/scripts/rundemo.sh:36]: Using cirros didn't work? Maybe its better to remove the lines then instead of commenting them out.  Now it looks like people can use cirros  instead of the ubuntu images below if they want to.

File: build/IPDK_Container/README_HOST_INSTALL.md
Comment (stolsma) [inline build/IPDK_Container/README_HOST_INSTALL.md:71]: Shouldn't the new `rundemo.sh` script being called here instead of the above commands?

**Review by stolsma — CHANGES_REQUESTED** [review]
Found some small things that need action I think before merge. Also have some questions,  see comments. In general looks very good!

**Comment by stolsma** [comment]
Dont know what the policy on merging pull req is but will you be squashing the commits to one commit before merging? Just asking for my own pull req that I'm working on. (Do I need to tidy the intermediate commits,  as I'm doing now, or is everything going in one commit in the end😀)

File: build/IPDK_Container/README_HOST_INSTALL.md
Comment (mestery) [reply build/IPDK_Container/README_HOST_INSTALL.md:62]: Yes.

File: build/IPDK_Container/scripts/rundemo.sh
Comment (mestery) [reply build/IPDK_Container/scripts/rundemo.sh:36]: Yes, cirrus had issues with cloud-init. I will remove, good catch.

File: build/IPDK_Container/scripts/rundemo.sh
Comment (mestery) [reply build/IPDK_Container/scripts/rundemo.sh:44]: Good catch! I'll remove the steps to move back and forth from raw.

File: build/IPDK_Container/scripts/rundemo.sh
Comment (mestery) [reply build/IPDK_Container/scripts/rundemo.sh:135]: I recall needing this during testing, but your points are valid. I'll remove and re-run locally and verify it works still. I think the real issue was the shared hugepages mount you mentioned in #46.

File: build/IPDK_Container/scripts/rundemo.sh
Comment (mestery) [reply build/IPDK_Container/scripts/rundemo.sh:143]: Yup, will remove.

File: build/IPDK_Container/scripts/set_hugepages.sh
Comment (mestery) [reply build/IPDK_Container/scripts/set_hugepages.sh:13]: Running mount by hand doesn't automatically update /etc/fstab.

Comment (stolsma) [reply build/IPDK_Container/scripts/set_hugepages.sh:13]: No, but why do you need to add it to fstab (that is only used at startup) if the check if the mount is there is always done at run of the script? (and if its not there it will be added...)

Comment (mestery) [reply build/IPDK_Container/scripts/set_hugepages.sh:13]: Ah, since this is running in a persistent Vagrant VM, the check is there because if you reload the Vagrant VM, it will exist in /etc/fstab and be automatically mounted, and both conditions here will work correctly.

**Comment by mestery** [comment]
> Dont know what the policy on merging pull req is but will you be squashing the commits to one commit before merging? Just asking for my own pull req that I'm working on. (Do I need to tidy the intermediate commits, as I'm doing now, or is everything going in one commit in the end😀)

I don't think there is an official policy, I've already merged un-squashed commits, so I think that's fine as well for now.

**Comment by mestery** [comment]
@stolsma Let me know your thoughts on the latest commit, I'd like to merge this if you're ok, I fixed the most obvious problem, which was the hugepages issues you found. Thanks!

**Comment by stolsma** [comment]
Looks good enough! Lgtm!

**Pull request merged**
